pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/realtime/PinotLLCRealtimeSegmentManager.java

```diff
@@ -75,26 +82,114 @@ public static PinotLLCRealtimeSegmentManager getInstance() {
    * The topic name is being used as a dummy helix resource name. We do not read or write to zk in this
    * method.
    */
-  public ZNRecord assignKafkaPartitions(final String topicName, int nPartitions, final List<String> instanceNames, int nReplicas) {
-    final String resourceName = topicName;
-
-    List<String> partitions = new ArrayList<>(nPartitions);
-    for (int i = 0; i < nPartitions; i++) {
-      partitions.add(Integer.toString(i));
+  public void setupHelixEntries(final String topicName, final String realtimeTableName, int nPartitions,
+      final List<String> instanceNames, int nReplicas, long startOffset) {
+    if (nReplicas > instanceNames.size()) {
+      throw new RuntimeException("Replicas requested(" + nReplicas + ") cannot fit within number of instances(" +
+          instanceNames.size() + ") for table " + realtimeTableName + " topic " + topicName);
     }
+    /*
+     Cannot do this because of https://issues.apache.org/jira/browse/HELIX-631
```

Comment:
I'd expand on this comment to say that due to a bug in autorebalance we balance manually, see the helix bug.